hpx/runtime/serialization/detail/non_default_constructible.hpp

```diff
@@ -0,0 +1,31 @@
+//  Copyright (c) 2017 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0.
+//  See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_SERIALIZATION_NON_DEFAULT_CONSTRUCTIBLE_HPP
+#define HPX_SERIALIZATION_NON_DEFAULT_CONSTRUCTIBLE_HPP
+
+#include <hpx/config.hpp>
+
+#include <memory>
+
+namespace hpx { namespace serialization { namespace detail
+{
+    ///////////////////////////////////////////////////////////////////////////
+    // default fall-backs for serializing non-default-constructible types
+    template <typename Archive, typename T>
+    void save_construct_data(Archive&, T const*, unsigned)
+    {
```

Comment:
Symmetry and compatibility with Boost serialization. The rationale is that that the `serialize` function should be used to serialize the internal data of a class and the `*_construct_data` are used to serialize any external data that might be used for constructing an object of that class, so that containers would be able to deserialize them.